ecommerce/extensions/payment/tests/test_views.py

```
@@ -169,3 +172,87 @@ def test_invalid_signature(self):
+
+        self.basket = factories.create_basket()
+        self.basket.owner = factories.UserFactory()
+        self.basket.freeze()
+
+        self.processor = Paypal()
+        self.processor_name = self.processor.NAME
+
+        # Dummy request from which an HTTP Host header can be extracted during
+        # construction of absolute URLs
+        self.request = RequestFactory().post('/')
+
+    @httpretty.activate
+    def test_payment_execution(self):
+        """Verify that a user who has approved payment is redirected to the configured receipt page."""
+        self.mock_oauth2_response()
+
+        # Create a payment record the view can use to retrieve a basket
+        self.mock_payment_creation_response(self.basket)
+        self.processor.get_transaction_parameters(self.basket, request=self.request)
+
+        self.mock_payment_creation_response(self.basket, find=True)
+        self.mock_payment_execution_response(self.basket)
+        response = self.client.get(reverse('paypal_execute'), self.RETURN_DATA)
+        self.assertRedirects(
+            response,
+            u'{}?basket_id={}'.format(self.processor.receipt_url, self.basket.id),
+            fetch_redirect_response=False
+        )
+
+    @httpretty.activate
+    def test_payment_error(self):
+        """
+        Verify that a user who has approved payment is redirected to the configured receipt page when payment
+        execution fails.
+        """
+        self.mock_oauth2_response()
+
+        # Create a payment record the view can use to retrieve a basket
+        self.mock_payment_creation_response(self.basket)
+        self.processor.get_transaction_parameters(self.basket, request=self.request)
+
+        self.mock_payment_creation_response(self.basket, find=True)
+        self.mock_payment_execution_response(self.basket)
+        with mock.patch.object(PaypalPaymentExecutionView, 'handle_payment', side_effect=PaymentError):
```

How does one verify that a caught exception has been raised? `assertRaises` doesn't seem to be useful here, since the exception is swallowed before the code under test returns.

This situation is due to the fact that we're always taking the same action (i.e., redirecting the user) regardless of errors. Would a comment on the mocked methods suffice?